src/directives/view.js

```diff
@@ -0,0 +1,296 @@
+// @TODO none of the animation code in this directive has been tested. Not sure if it can be at this stage This needs further investigation.
+// @TODO this code does too much, it should be refactored, but I need to get this out ASAP
+angular.module('bicker_router').directive('view', function($log, $compile, $controller, ViewBindings, $q, State, $rootScope, $animate, $timeout, $injector, PendingViewCounter, $templateRequest, Route) {
+  let directive = {
+    restrict: 'E',
+    scope: false,
+    replace: true,
+    template: '<div></div>',
+
+    link(viewDirectiveScope, iElement, iAttrs) {
+      let viewCreated = false;
+      let viewScope = undefined;
+      let viewManagementPending = false;
+      let view = ViewBindings.getView(iAttrs.name);
+      let bindings = view.getBindings();
+
+      iElement.addClass('ng-hide');
+
+      let previousBoundState = undefined;
+      let previousBinding = undefined;
+
+      let getStateDataForBinding = binding => _.cloneDeep(State.getSubset(getStateFieldsFromBinding(binding)));
+
+      let getComponentFromBinding = function(binding, field) {
+        if (field == null) { field = 'component'; }
+        let source = binding[field] ? $injector.get(binding[field] + 'Directive')[0] : binding;
+        return _.defaults(_.pick(source, ['controller', 'templateUrl', 'controllerAs']), { controllerAs: '$ctrl' });
+      };
+
```

Comment:
Can you replace all of the "let xxx = function()" with "function xxx() { }" - I understand they're not quite the same with regard to hoisting, but it shouldn't pose problems for the most part.

Also please review usage of const vs let - if you bring in the eslint rules all of that should get picked up.